test/integration/server-selection/readpreference.test.ts

```diff
@@ -145,219 +149,171 @@ describe('ReadPreference', function () {
+  //         client.close(done);
+  //       });
+  //     });
+  //   }
+  // });
+
+  // TODO(NODE-7219): Remove test. Type safety is now enforced by TypeScript.
+  // it('Should correctly pass readPreferences specified as objects to collection methods', {
+  //   metadata: { requires: { mongodb: '>=2.6.0', topology: ['single', 'ssl'] } },
+  //
+  //   test: function (done) {
+  //     const configuration = this.configuration;
+  //     const client = configuration.newClient(configuration.writeConcernMax(), { maxPoolSize: 1 });
+  //     client.connect(function (err, client) {
+  //       const db = client.db(configuration.db);
+  //       expect(err).to.not.exist;
+  //       // Create read preference object.
+  //       const mySecondaryPreferred = { mode: 'secondaryPreferred', tags: [] };
+  //       const cursor = db.collection('test').find({}, { readPreference: mySecondaryPreferred });
+  //       cursor.toArray(function (err) {
+  //         expect(err).to.not.exist;
+  //         client.close(done);
+  //       });
+  //     });
+  //   }
+  // });
 
   it('Should correctly pass readPreferences on the Collection to listIndexes', {
-    metadata: { requires: { mongodb: '>=2.6.0', topology: ['single', 'ssl'] } },
+    metadata: { requires: { topology: ['single'] } },
 
-    test: function (done) {
-      const configuration = this.configuration;
-      const client = configuration.newClient(configuration.writeConcernMax(), { maxPoolSize: 1 });
-      client.connect(function (err, client) {
-        const db = client.db(configuration.db);
-        expect(err).to.not.exist;
-        const cursor = db
-          .collection('test', { readPreference: ReadPreference.SECONDARY_PREFERRED })
-          .listIndexes();
-        test.equal(cursor.readPreference.mode, 'secondaryPreferred');
-        client.close(done);
-      });
+    test: async function () {
+      const db = client.db(this.configuration.db);
+      const cursor = db
+        .collection('test', { readPreference: ReadPreference.SECONDARY_PREFERRED })
+        .listIndexes();
+      test.equal(cursor.readPreference.mode, 'secondaryPreferred');
     }
   });
 
-  it('Should throw an error on an invalid readPreference', function (done) {
-    const configuration = this.configuration;
-
-    const client = configuration.newClient();
-    client.connect((err, client) => {
-      const db = client.db(configuration.db);
-      expect(db.collection.bind(db, 'test', { readPreference: 'invalid' })).to.throw(
-        'Invalid read preference mode "invalid"'
-      );
-
-      client.close(done);
-    });
+  it('Should throw an error on an invalid readPreference', async function () {
+    const db = client.db(this.configuration.db);
+    expect(db.collection.bind(db, 'test', { readPreference: 'invalid' })).to.throw(
+      'Invalid read preference mode "invalid"'
+    );
   });
 
+  // TODO(NODE-7219): Hedge reads are deprecated in MongoDB Server version 8.0.
```

Comment:
```suggestion
```

Reply:
Suggest removing this; if/when hedged reads are ever removed from the server, this test will fail and we can add a runOnRequirement at that time.